src/js/saspowerschoolff.js

```diff
@@ -188,6 +203,85 @@ function login_page () {
     });
 }
 
+function calculate_cumulative_gpa (current_courses) {
+    const list_of_gpas = [];
+    const all_courses = [];
+    const credit_hour_list = [];
+    let element_list = [];
+    const fetches = [];
+    // Fetches grade history page
+    const gpas = fetch("https://powerschool.sas.edu.sg/guardian/termgrades.html")
+        .then(response => response.text())
+        .then(data => {
+            const el = document.createElement("html");
+            el.innerHTML = data;
+            const tabs = el.getElementsByClassName("tabs")[0].getElementsByTagName("li");
+            for (let i = 0; i < tabs.length; i++) {
```

Comment:
```suggestion
            // Iterate until the end of tabs or until no longer at a high school semester.
            for (let i = 0; i < tabs.length && /HS$/.test(tabs.innerText); i++) {
```
To make sure that middle school grades are not included.

Reply:
Oh didn't think of that, thanks.

Reply:
Does this work for you? I just tested this and it results in my cumulative being calculate incorrectly.

Reply:
Just needed to change it to tabs[i].innerText.